magefiles/main.go

```diff
@@ -14,27 +15,101 @@ import (
 
 // BootstrapTools installs all tools needed tobuild and release Armada.
 // For the list of tools this will install, see tools.yaml in the root directory
+type ToolsList struct {
+	Tools []string
+}
+
 func BootstrapTools() error {
 	mg.Deps(goCheck)
 	type ToolsList struct {
 		Tools []string
 	}
 
-	tools := &ToolsList{}
-	err := readYaml("tools.yaml", tools)
+	if isGitHubActions() {
+		gomodcache, err := os.MkdirTemp("", "gomodcache")
+		if err != nil {
+			fmt.Println("Error creating temporary GOMODCACHE directory:", err)
+			return err
+		}
+		defer os.RemoveAll(gomodcache)
+
+		gocache, err := os.MkdirTemp("", "gocache")
+		if err != nil {
+			fmt.Println("Error creating temporary GOCACHE directory:", err)
+			return err
+		}
+		defer os.RemoveAll(gocache)
+
+		os.Setenv("GOMODCACHE", gomodcache)
+		os.Setenv("GOCACHE", gocache)
```

Comment:
these environment variables should _only_ be set for the `go install` commands and not for the current process